### PR TITLE
fn-cli: init at 0.6.6

### PR DESCRIPTION
**Pull request by bolasblack**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Added a release notes entry if the change is major or breaking
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by bolasblack** [comment]
This PR is recreated from PR #92924 (I finally have time to fix my PR 😝 )

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 124447` at 21bad170 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>fn-cli</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/no-uri-literals.md">no-uri-literals</a>

URI literals are deprecated.
Near pkgs/applications/networking/cluster/fn-cli/default.nix:11:16:
```
   |
11 |     homepage = https://fnproject.io;
   |                ^
```
</li>
  </ul>
</details>

File: pkgs/applications/networking/cluster/fn-cli/default.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
+{ lib, buildGoModule, fetchFromGitHub, docker }:
+
+buildGoModule rec {
+  pname = "fn";
+  version = "0.6.6";
+
+  name = "${pname}-${version}";
```
Comment (legendofmiracles) [inline pkgs/applications/networking/cluster/fn-cli/default.nix:7]: This is implicitly set

File: pkgs/applications/networking/cluster/fn-cli/default.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+  meta = with lib; {
+    description = "Command-line tool for the fn project";
+    homepage = "https://fnproject.io";
+    license = licenses.asl20;
+    maintainers = [ maintainers.c4605 ];
+  };
```
Comment (legendofmiracles) [inline pkgs/applications/networking/cluster/fn-cli/default.nix:14]: Usually goes to the end of the file

File: pkgs/applications/networking/cluster/fn-cli/default.nix [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+  src = fetchFromGitHub {
+    owner = "fnproject";
+    repo = "cli";
+    rev = "${version}";
```
Comment (legendofmiracles) [inline pkgs/applications/networking/cluster/fn-cli/default.nix:19]:
```suggestion
    rev = version";
```

**Comment by bolasblack** [comment]
Hi @legendofmiracles, thanks for your suggestion, I have resolved all the problems 😁

**Review by legendofmiracles — APPROVED** [review]

**Pull request merged**